src/main/flight/pid.c

```diff
@@ -1299,44 +1318,42 @@ void FAST_CODE pidController(const pidProfile_t *pidProfile, const rollAndPitchT
             if (cmpTimeUs(currentTimeUs, levelModeStartTimeUs) > CRASH_RECOVERY_DETECTION_DELAY_US) {
                 detectAndSetCrashRecovery(pidProfile->crash_recovery, axis, currentTimeUs, delta, errorRate);
             }
-#if defined(USE_D_CUT)
-            if (dtermCutPercent){
-                dtermCutFactor = dtermCutRangeApplyFn((filter_t *) &dtermCutRange[axis], delta);
-                dtermCutFactor = fabsf(dtermCutFactor) * dtermCutGain;
-                dtermCutFactor = dtermCutLowpassApplyFn((filter_t *) &dtermCutLowpass[axis], dtermCutFactor);
-                dtermCutFactor = MIN(dtermCutFactor, 1.0f);
-                dtermCutFactor = dtermCutPercentInv + (dtermCutFactor * dtermCutPercent);
-            }
-            if (axis == FD_ROLL) {
-                DEBUG_SET(DEBUG_D_CUT, 2, lrintf(pidCoefficient[axis].Kd * dtermCutFactor * 10.0f / DTERM_SCALE));
-            } else if (axis == FD_PITCH) {
-                DEBUG_SET(DEBUG_D_CUT, 3, lrintf(pidCoefficient[axis].Kd * dtermCutFactor * 10.0f / DTERM_SCALE));
+#if defined(USE_D_MIN)
+            if (dMinPercent[axis]) {
+                float dMinGyroFactor = dMinRangeApplyFn((filter_t *) &dMinRange[axis], delta);
+                dMinGyroFactor = fabsf(dMinGyroFactor) * dMinGyroGain;
+                const float dMinSetpointFactor = (fabsf(pidSetpointDelta)) * dMinSetpointGain;
+                dMinFactor = MAX(dMinGyroFactor, dMinSetpointFactor);
+                dMinFactor = dMinPercent[axis] + (1.0f - dMinPercent[axis]) * dMinFactor;
+                dMinFactor = dMinLowpassApplyFn((filter_t *) &dMinLowpass[axis], dMinFactor);
+                dMinFactor = MIN(dMinFactor, 1.0f);
+                if (axis == FD_ROLL) {
+                    DEBUG_SET(DEBUG_D_MIN, 0, lrintf(dMinGyroFactor * 100.0f));
+                    DEBUG_SET(DEBUG_D_MIN, 1, lrintf(dMinSetpointFactor * 100.0f));
+                    DEBUG_SET(DEBUG_D_MIN, 2, lrintf(pidCoefficient[axis].Kd * dMinFactor * 10.0f / DTERM_SCALE));
+                } else if (axis == FD_PITCH) {
+                    DEBUG_SET(DEBUG_D_MIN, 3, lrintf(pidCoefficient[axis].Kd * dMinFactor * 10.0f / DTERM_SCALE));
+                }
+                pidData[axis].D = pidCoefficient[axis].Kd * delta * tpaFactor * dMinFactor;
```

Comment:
This should be removed - duplicate D calculation.

Reply:
I think it needs to be here since sMinFactor depends on USE_D_MIN to be defined.
Used and ifndef USE_D_MIN for the D calculation when USE_D_MIN isn't defined.
Hopefully this is OK?

Reply:
This is okay also. The problem before was that axis D calculation would have happened twice when `USE_D_MIN` was defined. Now it's okay.

I realize that @fujin steered you the way it is now, but I would have stayed with a single D calculation that included `* dtermCutFactor` after the block and initialized `float dtermCutFactor = 1.0f` above the block. The reason being that it keeps a single D calculation rather than having it replicated (slightly differently) in two places. With multiple instances it makes it more likely to be missed for future changes to the D calculation.

But the tradeoff is that yes it's (very) slightly slower for the unnecessary multiplication * 1.0 for the `USE_D_MIN` undefined case. So I personally would have given up a small performance hit for some F3's (the only place where it's undefined) for future maintainability.

Don't change it yet, let's have some comments from others.

Reply:
Actually this little bit of mayhem is entirely my doing... and I didn't really know what the best way would be.  I'm happy to be guided by your advice.

Reply:
The reason I suggested the duplicated form was to not leak the variables from D_MIN ifdef. I'm okay with condensing it to one and simply overriding the final factor.

Reply:
Just to close out this half with what we discussed on Slack - the compiler will optimize out the redundant multiplication, so the dMinFactor will be initialized to 1.0f outside of the ifdef, then overriden inside of it. The only concern now is the name is not so good (leaks dMin details. perhaps `dtermScale dtermMultiplier dScaleFactor` etc)

Reply:
Resolved after discussions in favour of keeping the D_Min multiplier of 1.0 outside the ifdef D_MIN code, so that there is only one single line where D itself is calculated.